pkg/kanctl/repositoryserver.go

```diff
@@ -0,0 +1,292 @@
+	if err != nil {
+		return nil, err
+	}
+	repositoryServerUserAccessSecret, err := cli.CoreV1().Secrets(rsParams.namespace).Get(ctx, rsParams.repositoryServerUserAccess, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	repositoryServerAdminUserAccessSecret, err := cli.CoreV1().Secrets(rsParams.namespace).Get(ctx, rsParams.repositoryServerAdminUserAccess, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	repositoryPasswordSecret, err := cli.CoreV1().Secrets(rsParams.namespace).Get(ctx, rsParams.repositoryPassword, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	locationSecret, err := cli.CoreV1().Secrets(rsParams.namespace).Get(ctx, rsParams.location, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	locationCredsSecret, err := cli.CoreV1().Secrets(rsParams.namespace).Get(ctx, rsParams.locationCreds, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	return &v1alpha1.RepositoryServer{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: `kopia-repo-server-`,
+		},
+		Spec: v1alpha1.RepositoryServerSpec{
+			Storage: v1alpha1.Storage{
+				SecretRef: corev1.SecretReference{
+					Name:      locationSecret.GetName(),
+					Namespace: locationSecret.GetNamespace(),
+				},
+				CredentialSecretRef: corev1.SecretReference{
+					Name:      locationCredsSecret.GetName(),
+					Namespace: locationCredsSecret.GetNamespace(),
+				},
+			},
+			Repository: v1alpha1.Repository{
+				RootPath: rsParams.prefix,
+				Username: rsParams.repositoryUser,
+				Hostname: defaultRepositoryServerHost,
+				PasswordSecretRef: corev1.SecretReference{
+					Name:      repositoryPasswordSecret.GetName(),
+					Namespace: repositoryPasswordSecret.GetNamespace(),
+				},
+			},
+			Server: v1alpha1.Server{
+				UserAccess: v1alpha1.UserAccess{
+					UserAccessSecretRef: corev1.SecretReference{
+						Name:      repositoryServerUserAccessSecret.GetName(),
+						Namespace: repositoryServerUserAccessSecret.GetNamespace(),
+					},
+					Username: rsParams.repositoryServerUser,
+				},
+				AdminSecretRef: corev1.SecretReference{
+					Name:      repositoryServerAdminUserAccessSecret.GetName(),
+					Namespace: repositoryServerAdminUserAccessSecret.GetNamespace(),
+				},
+				TLSSecretRef: corev1.SecretReference{
+					Name:      tlsSecret.GetName(),
+					Namespace: tlsSecret.GetNamespace(),
+				},
+			},
+		},
+	}, nil
+}
+
+func waitForRepositoryServerReady(ctx context.Context, cli *versioned.Clientset, rs *v1alpha1.RepositoryServer) error {
+	timeoutCtx, waitCancel := context.WithTimeout(ctx, contextWaitTimeout)
+	defer waitCancel()
+	err := poll.Wait(timeoutCtx, func(ctx context.Context) (bool, error) {
+		repositoryServer, err := cli.CrV1alpha1().RepositoryServers(rs.GetNamespace()).Get(ctx, rs.GetName(), metav1.GetOptions{})
+		if repositoryServer.Status.Progress == v1alpha1.ServerReady && err == nil {
+			return true, nil
+		}
+		return false, nil
```

Comment:
What would the kanctl command return  `repositoryServer.status.Progress==v1alpha1.ServerStopped`? do we want to return error to the user?

Reply:
What could we do in this case @kale-amruta ?

Reply:
Right now it will wait till kopia repo server is in ready state, and if any other state, it would return error after timeout